docker/Dockerfile

```diff
@@ -0,0 +1,39 @@
+# syntax = docker/dockerfile:1.2
+
+FROM debian:buster-slim
+
+LABEL maintainer="BentoML Contributor <contact@bentoml.ai>"
+
+ENV DOCKER_BUILDKIT=1
+
+RUN apt-get update && \
+    apt-get install -y python3 python3-pip bash curl build-essential
+
+SHELL ["/bin/bash", "-exo", "pipefail", "-c"]
+
+# https://jpetazzo.github.io/2015/09/03/do-not-use-docker-in-docker-for-ci/
```

Comment:
I think for the given need of having an environment up, docker in docker should not be too much of an issue